components/brave_rewards/resources/extension/brave_rewards/content.ts

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Utils
+import { getMessage } from './background/api/locale_api'
+
+const getTweetMetaData = (tweet: Element): RewardsDonate.TweetMetaData | null => {
+  if (!tweet) {
+    return null
+  }
+  const tweetTextElements = tweet.getElementsByClassName('tweet-text')
+  if (!tweetTextElements || tweetTextElements.length === 0) {
+    return null
+  }
+  return {
+    name: tweet.getAttribute('data-name') || '',
+    screenName: tweet.getAttribute('data-screen-name') || '',
+    userId: tweet.getAttribute('data-user-id') || '',
+    tweetText: tweetTextElements[0].textContent || ''
+  }
+}
+
+const createBraveTipAction = (tweet: Element) => {
+  // Create the tip action
+  const braveTipAction = document.createElement('div')
```

Comment:
Right now I believe this method of creating DOM elements has the following issues that were brought up in an older security discussion:
- The site can detect that the user is a Brave Rewards user via detecting the element
- The site can intercept the click and do something malicious
- The site's styles can unintentionally / intentionally interfere with the button.

These issues can be at least minimized by at least creating a single element, and then putting all the button content inside that element's shadow DOM.

And then, we could explore not having to add an element at all, but instead adding the button to the parent's closed shadow DOM. This is what I did on my prototype for the popup.

Let me know what you think @emerick

Reply:
Really good point, will get that in there!

Reply:
@petemill Do you happen to know: is there a clever way to access a CSS class defined in the parent document from the shadow DOM? The issue I'm running into is that we reference a handful of CSS classes defined in the Twitter stylesheet, but the shadow DOM doesn't have access to those styles. I'm reluctant to copy those underlying styles directly into the shadow DOM as then it becomes kind of brittle (e.g., if/when Twitter tweaks those styles).

Reply:
No, I guess that's the point of the shadow DOM - that the document styles _cannot_ affect the shadow elements. So it's a trade-off between wanting to affect the styles intentionally and not wanting unintentional or malicious styles. Perhaps someone like @diracdeltas could weight in on the priority.

However, I would add that using the css class names could be just as brittle, since these could change anytime. One way to somewhat get around that (again not using shadow DOM) would be to find an existing button's node that is a sibling of our 'tip' button, and call `.cloneNode()` on it, then add our own (inline) styles to override.

Are there many / any native document styles that we need to inherit anyway, since AFAIK our button is simply our icon?

Reply:
Right, I see what you're saying (and you make a good point about the class names). I think you're right that we don't end up needing very many of the inherited styles, just some text properties for the "Tip" label and sizing properties most likely. Probably not too much harm including them in the shadow DOM if that's the case. I'll experiment with that, thanks again!

Reply:
++ for encapsulating our injected element in a shadow DOM

Reply:
Added to shadow DOM.